codecs/mozjpeg_enc/Makefile

```diff
@@ -0,0 +1,45 @@
+CODEC_DIR := node_modules/mozjpeg
+CODEC_OUT_RELATIVE := .libs/libjpeg.a rdswitch.o
+CODEC_OUT := $(addprefix $(CODEC_DIR)/, $(CODEC_OUT_RELATIVE))
+OUT_JS := mozjpeg_enc.js
+OUT_WASM := $(OUT_JS:.js=.wasm)
+
+.PHONY: all clean
+
+all: $(OUT_JS)
+
+%.js: %.cpp $(CODEC_OUT)
+	$(CXX) \
+		-I $(CODEC_DIR) \
+		${CXXFLAGS} \
+		${LDFLAGS} \
+		--bind \
+		--closure 1 \
+		-s ALLOW_MEMORY_GROWTH=1 \
+		-s MODULARIZE=1 \
+		-s 'EXPORT_NAME="$(basename $(@F))"' \
+		-o $@ \
+		$+
+
+# This one is a bit special: there is no rule for .libs/libjpeg.a
+#  so we use libjpeg.la which implicitly builds that one instead.
+$(CODEC_DIR)/.libs/libjpeg.a: $(CODEC_DIR)/Makefile
+	$(MAKE) -C $(CODEC_DIR) libjpeg.la
+
+$(CODEC_DIR)/rdswitch.o: $(CODEC_DIR)/Makefile
+	$(MAKE) -C $(CODEC_DIR) rdswitch.o
+
+$(CODEC_DIR)/Makefile: $(CODEC_DIR)/configure
+	cd $(CODEC_DIR) && ./configure \
+		--disable-shared \
+		--without-turbojpeg \
+		--without-simd \
+		--without-arith-enc \
```

Comment:
@surma By the way, just to double-check here - I've disabled few more options/modes that we don't use, and, as far as I can tell from their description, we don't want to use.

This shaved off another ~9.5 KB off the Wasm binary, but I want to make sure this is okay as I don't understand image codecs that well :)

Reply:
@surma Ping.

Reply:
Oh yeah, that’s totally fine by me if it still works :D

Reply:
Well I *hope* it does :D